src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -32,14 +32,27 @@ public partial class JsonTypeInfo
         // Add method delegate for non-generic Stack and Queue; and types that derive from them.
         internal object? AddMethodDelegate { get; set; }
 
-        internal JsonPropertyInfo? DataExtensionProperty { get; private set; }
+        internal JsonPropertyInfo? DataExtensionProperty { get; set; }
 
         // If enumerable or dictionary, the JsonTypeInfo for the element type.
         private JsonTypeInfo? _elementTypeInfo;
 
         // Avoids having to perform an expensive cast to JsonTypeInfo<T> to check if there is a Serialize method.
         internal bool HasSerialize { get; set; }
```

Comment:
This sentinel could now be removed, replaced with a null check to the `Serialize` method.

Reply:
I will see if that's the case

Reply:
possibly, I'd prefer this change separate because we currently have couple of code path which need to transition from non-generic to generic and I need to find a good way to do that without sacrificing perf so I'd rather we've done this separately